src/plugins/plot/vue/single/MctTicks.vue

```diff
@@ -0,0 +1,245 @@
+            if (!range) {
+                delete this.min;
+                delete this.max;
+                delete this.interval;
+                delete this.tickRange;
+                delete this.ticks;
+                delete this.shouldCheckWidth;
+
+                return;
+            }
+
+            const format = this.axis.get('format');
+            if (!format) {
+                return;
+            }
+
+            this.min = range.min;
+            this.max = range.max;
+            this.interval = Math.abs(range.min - range.max);
+            if (this.shouldRegenerateTicks(range, forceRegeneration)) {
+                let newTicks = this.getTicks();
+                this.tickRange = {
+                    min: Math.min.apply(Math, newTicks),
+                    max: Math.max.apply(Math, newTicks),
+                    step: newTicks[1] - newTicks[0]
+                };
+
+                newTicks = newTicks
+                    .map(function (tickValue) {
+                        return {
+                            value: tickValue,
+                            text: format(tickValue)
+                        };
+                    }, this);
+
+                if (newTicks.length && typeof newTicks[0].text === 'string') {
+                    const tickText = newTicks.map(function (t) {
+                        return t.text;
+                    });
+                    const prefix = tickText.reduce(commonPrefix);
+                    const suffix = tickText.reduce(commonSuffix);
+                    newTicks.forEach(function (t) {
+                        t.fullText = t.text;
+                        if (suffix.length) {
+                            t.text = t.text.slice(prefix.length, -suffix.length);
+                        } else {
+                            t.text = t.text.slice(prefix.length);
+                        }
+                    });
+                }
+
+                this.ticks = newTicks;
+                this.shouldCheckWidth = true;
+            }
+
+            this.scheduleTickUpdate();
+        },
+
+        scheduleTickUpdate() {
+            if (this.tickUpdate) {
+                return;
+            }
+
+            this.tickUpdate = true;
+            setTimeout(this.doTickUpdate.bind(this), 0);
+        },
+
+        doTickUpdate() {
+            if (this.shouldCheckWidth) {
+                // TODO: Do we need to do something for these 2 $digests?
+                // this.$scope.$digest();
+                const tickElements = this.$refs.tickContainer.querySelectorAll('.gl-plot-tick > span');
+
+                const tickWidth = Number([].reduce.call(tickElements, function (memo, first) {
+                    return Math.max(memo, first.offsetWidth);
+                }, 0));
+
+                this.tickWidth = tickWidth;
+                this.$emit('plotTickWidth', tickWidth);
+                this.shouldCheckWidth = false;
+            }
+
+            // this.$scope.$digest();
```

Comment:
Angular had some weird irregularities. We wont need any forced digests with vue

Reply:
Yeah, I'll remove those comments.